src/Core/Authoring/PhysicalPackageFile.cs

```diff
@@ -89,9 +91,171 @@ public IEnumerable<FrameworkName> SupportedFrameworks
+            private static uint GetAccess(FileAccess access)
+            {
+                uint uaccess = 0;
+                switch (access)
+                {
+                    case FileAccess.Read:
+                        uaccess = GENERIC_READ;
+                        break;
+                    case FileAccess.ReadWrite:
+                        uaccess = GENERIC_READ | GENERIC_WRITE;
+                        break;
+                    case FileAccess.Write:
+                        uaccess = GENERIC_WRITE;
+                        break;
+                }
+                return uaccess;
+            }
+
+            // uint GetShare(FileShare share)
+            // Converts the FileShare constant to win32 constant
+            private static uint GetShare(FileShare share)
+            {
+                uint ushare = 0;
+                switch (share)
+                {
+                    case FileShare.Read:
+                        ushare = FILE_SHARE_READ;
+                        break;
+                    case FileShare.ReadWrite:
+                        ushare = FILE_SHARE_READ | FILE_SHARE_WRITE;
+                        break;
+                    case FileShare.Write:
+                        ushare = FILE_SHARE_WRITE;
+                        break;
+                    case FileShare.Delete:
+                        ushare = FILE_SHARE_DELETE;
+                        break;
+                    case FileShare.None:
+                        ushare = 0;
+                        break;
+
+                }
+                return ushare;
+            }
+
+            public static FileStream Open(string filepath, FileMode mode, FileAccess access, FileShare share)
+            {
+                //opened in the specified mode , access and  share
+                FileStream fs = null;
+                uint umode = GetMode(mode);
+                uint uaccess = GetAccess(access);
+                uint ushare = GetShare(share);
+                if (mode == FileMode.Append)
+                    uaccess = FILE_APPEND_DATA;
+                // If file path is disk file path then prepend it with \\?\
+                // if file path is UNC prepend it with \\?\UNC\ and remove \\ prefix in unc path.
+                if (filepath.StartsWith(@"\\"))
+                {
+                    filepath = @"\\?\UNC\" + filepath.Substring(2, filepath.Length - 2);
+                }
+                else
+                    filepath = @"\\?\" + filepath;
+                SafeFileHandle sh = CreateFileW(filepath, uaccess, ushare, IntPtr.Zero, umode, FILE_ATTRIBUTE_NORMAL, IntPtr.Zero);
+                int iError = Marshal.GetLastWin32Error();
+                if ((iError > 0 && !(mode == FileMode.Append && iError != ERROR_ALREADY_EXISTS)) || sh.IsInvalid)
+                    throw new Exception("Error opening file Win32 Error:" + iError);
+
+                fs = new FileStream(sh, access);

```

Comment:
The next line can throw and you don't get a disposable back. You want to guard against it